packages/shared/scope-hoisting/src/link.js

```diff
@@ -399,8 +400,18 @@ export function link({
             .getDependencies(mapped)
             .find(dep => dep.moduleSpecifier === source.value)
         );
-        let mod = nullthrows(bundleGraph.getDependencyResolution(dep));
-        path.replaceWith(t.valueToNode(mod.id));
+        if (!bundleGraph.getDependencyResolution(dep)) {
+          // was excluded from bundling
+          path.replaceWith(
+            REQUIRE_RESOLVE_CALL_TEMPLATE({ID: t.stringLiteral(source.value)})
```

Comment:
Should we only do this with CommonJS outputFormat? We should probably also check if the specifier starts with `.` (i.e. local file?). Probably wouldn't work once bundled...

Reply:
> Should we only do this with CommonJS outputFormat?

Done

> We should probably also check if the specifier starts with

A local file would have a dependency resolution, and the else clause always throws?

Reply:
> Should we only do this with CommonJS outputFormat?

Done

> We should probably also check if the specifier starts with

A local file would have a dependency resolution, and the else clause always throws?